### DISCUSSION TITLE
Oqtane Application in MAUI

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#2004] [General]
Hi,

How do we Oqtane's components (module's components) can be hosted in MAUI without doing any further development for mobile apps iOS and Android?

Any steps or suggestions please?

Thanks & Regards!

**Comment by sbwalker** [comment]
This is an area that needs further research... if anyone has time...

**Reply by Sharafudeen** [reply]
I don't know whether my suggestions are right or not..

**Option-1:**
1. Make all Oqtane's Module DLLs in one package and referenced in MAUI. (Oqtane framework can be provided a module to package all module's DLLs in one package)
2. Register all module services with new service helper class in MAUI.

**Option-2:**
1. Copy all components & wwwroot assets (files & folders) from Oqtane Module1.
2. Create a folder with same name Module1 and paste Module1 components & wwwroot assets in MAUI.
3. Register all module services with new service helper class in MAUI.

**Option-3:**
1. Add MAUI project in Oqtane solution.
2. Add Oqtane framework project references to MAUI.
3. Register all module services with new service helper class in MAUI.

All options requires Oqtane framework DLLs (Client, Server, Shared) to reference in MAUI.
